pages/podcasts/_id.vue

```diff
@@ -4,18 +4,21 @@
 
 <script>
 import Podcast from '@/components/podcasts/Podcast'
+import { mapGetters } from 'vuex'
 
 export default {
+  middleware: 'redirect',
   components: { Podcast },
   data() {
-    return {
-      podcast: {
-        id: 1,
-        title: "Let's talk about Elixir!",
-        description: 'Lorem ipsum desu desu'
-      }
+    return {}
+  },
+  computed: {
+    ...mapGetters('podcasts', ['getPodcastById']),
+    podcast() {
+      return this.getPodcastById(this.$route.params.id)
```

Comment:
You can directly recieve params.id as a props in this component [see here](https://router.vuejs.org/guide/essentials/passing-props.html)